src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -1276,6 +1587,236 @@ public void GetAsync_ServerNeedsAuthAndNoCredential_StatusCodeUnauthorized()
             });
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        [InlineData(null)]
+        public async Task ReadAsStreamAsync_HandlerProducesWellBehavedResponseStream(bool? chunked)
+        {
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
+            {
+                var request = new HttpRequestMessage(HttpMethod.Get, uri);
+                using (var client = new HttpMessageInvoker(CreateHttpClientHandler()))
+                using (HttpResponseMessage response = await client.SendAsync(request, CancellationToken.None))
+                {
+                    using (Stream responseStream = await response.Content.ReadAsStreamAsync())
+                    {
+                        Assert.Same(responseStream, await response.Content.ReadAsStreamAsync());
+
+                        // Boolean properties returning correct values
+                        Assert.True(responseStream.CanRead);
+                        Assert.False(responseStream.CanWrite);
+                        Assert.False(responseStream.CanSeek);
+
+                        // Not supported operations
+                        Assert.Throws<NotSupportedException>(() => responseStream.BeginWrite(new byte[1], 0, 1, null, null));
```

Comment:
Beyond the scope of this PR, but...

I wish we had a general "test if this stream is well behaved" test, somewhere in the common test code.  There are a zillion different Stream implementations and it would be nice to test them all in a consistent, thorough manner.

Reply:
OK